2024/02-data-comm/slides.qmd

```diff
@@ -85,33 +89,274 @@ Use `theme_set(theme_bw())` outside your plot code to set the theme for a whole
 - Marks correspond to `geoms`
-    - `geom_point, geom_line, geom_col, ...`
-- Align marks with data using `aes`:
-    - `x` and `y` for position
-    - `color` and `fill` for color scale
-      - Discrete or continuous
-    - `alpha` for opacity
-    - `size` for size
-    
-
-## Example modifying geoms
-
-## Worked example as a group
\ No newline at end of file
+  - `geom_point, geom_line, geom_col, ...`
+- We map marks to data using `aes` (aesthetics):
+  - `x` and `y` for position
+  - `color` and `fill` for color scale
+  - Discrete or continuous
+  - `alpha` for opacity
+  - `size` for size
+- Aesthetics can be set manually, if not mapped to data. 
+
+
+## Counterintuitive marks and channels
+
+Size is much worse than position for showing continuous data!
+
+```{r, echo = T}
+
+ggplot(penguins, aes(sex, species, size = body_mass_g)) +
+geom_jitter() +
+theme(panel.background = element_blank())
+
+```
+
+
+## Counterintuitive marks and channels
+
+Size is much worse than position for showing continuous data!
+
+```{r, echo = T}
+
+ggplot(penguins, aes(body_mass_g)) +
+geom_histogram() +
+theme(panel.background = element_blank()) +
+facet_grid(rows = vars(species),
+cols = vars(sex), 
+switch = "y")
+
+```
+
+
+## Marks and channels: Options
+
+Shape is less effective than color for differentiating categories!
+
+```{r, echo = T}
+
+ggplot(penguins, aes(body_mass_g, 
+bill_depth_mm, shape = species)) +
+geom_point() +
+theme(panel.background = element_blank()) 
+
+```
+
+## Marks and channels: Options
+
+Shape is less effective than color for differentiating categories!
+
+
+```{r, echo = T}
+
+ggplot(penguins, aes(body_mass_g, 
+bill_depth_mm, color = species)) +
+geom_point() +
+theme(panel.background = element_blank()) 
+
+```
+
+## Effective color scales
+
+- There are multitudes of color palettes available for R.
+- Color palettes can be sequential, categorical, diverging, or bivariate.
+- Not all color palettes will be equally accessible to all viewers.
+- [Cols4All](https://mtennekes.github.io/cols4all/) helps you explore color palettes and find one to meet your needs. 
+
+## `cols4all`
+
+```{r, eval = F, echo = T}
+install.packages("cols4all", dependencies = TRUE)
+
+library(cols4all)
+
+c4a_gui()
+```
+
+```{r}
+library(cols4all)
+
+ggplot(penguins, aes(body_mass_g, 
+bill_depth_mm, color = species)) +
+geom_point() +
+theme(panel.background = element_blank()) +
+scale_color_discrete_c4a_cat(palette = "carto.vivid")
+```
+
+## A data-to-viz helper: Esquisse
+
+- [`esquisse`](https://dreamrs.github.io/esquisse/) is a ggplot2 *extension* for exploring data-viz pairings
+- Stay tuned for session 3 for more on extensions!
+
+
+## Esquisse demo
+
+```{r, echo = T, eval = F}
+install.packages("esquisse")
+
+esquisse::esquisser(penguins)
+```
+
+## Let's put it together
+
+- [Look at one of the example data tables here.](https://docs.google.com/spreadsheets/d/1U5OpOxM-aU8KV7xeqHaqsBoAV2koMYZYBCtNn7cCEFg/edit?gid=1285249029#gid=1285249029)
```

Comment:
It might be useful for time to just pick one of the datasets in advance and have everyone think about the same one?